server/swap/swap.go

```diff
@@ -516,10 +631,40 @@ func (s *Swapper) checkInaction(assetID uint32) {
 			return
 		}
 
-		failMatch := func(userAtFault account.AccountID) {
-			deletions = append(deletions, match.ID().String())
+		failMatch := func(makerFault bool) {
+			orderAtFault := match.Taker // an order.Order
+			if makerFault {
+				orderAtFault = match.Maker
+			}
+			deletions = append(deletions, match)
+
+			// Record the end of this match's processing.
 			s.storage.SetMatchInactive(db.MatchID(match.Match))
-			s.authMgr.Penalize(userAtFault, account.FailureToAct)
+
+			// Create the server-generated cancel order, and register it with
+			// the AuthManager for cancellation ratio computation.
+			coid, revTime, err := s.storage.RevokeOrder(orderAtFault)
+			if err == nil {
+				s.authMgr.RecordCancel(orderAtFault.User(), coid, orderAtFault.ID(), revTime)
+			} else {
+				log.Errorf("Failed to revoke order %v with a new cancel order: %v",
+					orderAtFault.UID(), err)
+			}
+
+			// That's one less active swap for this order, and a failure.
+			s.orders.swapFailure(orderAtFault)
```

Comment:
Good catch.  Yeah, we do need to decrement the swap count for the other.  Although I'm not sure how to handle crediting that other not-at-fault party with completion.  If that order was involved in no previous swaps but the order is off-book and not failed itself, I don't know that the user deserves the credit (note that we don't presently have a way to detect this case!).  But if the order was involved in previous swaps that they completed successfully, then this event should credit the user.  I'm not actually too concerned about crediting the user in *both* cases though, especially since it does sort of punish the user by not crediting them, and it legit users will always be seeking completions to give them more flexibility.

Reply:
Resolved in https://github.com/decred/dcrdex/pull/206/commits/3c726e65ef29828b850bc37914e805dd0312afd8
That does it for the scope of this PR, I think.